x/evm/types/journal.go

```diff
@@ -195,8 +195,21 @@ func (ch createObjectChange) revert(s *CommitStateDB) {
 		// perform no-op
 		return
 	}
+
 	// remove from the slice
-	s.stateObjects = append(s.stateObjects[:idx], s.stateObjects[idx+1:]...)
```

Comment:
we were deleting the object from the slice but we weren't deleting the item from the map nor updating the corresponding indexes of the other elements.

Reply:
this is probably the fix for the issue yesterday, can remove the `idx < len(csdb.stateObjects) ` check in `getStateObject` and `setStateObject` in #458

Reply:
thanks! let me add a few tests so that we can merge this to development